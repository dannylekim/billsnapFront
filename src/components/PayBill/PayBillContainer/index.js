import PayBillContainer from "./PayBillContainer";
import {connect} from "react-redux";
import {payABill} from "../../../redux/actions/billActions"

const mapDispatchToProps = (dispatch) => ({
    payABill: (amountPaid, billId) => dispatch(payABill(amountPaid, billId)),
});

export default connect(null, mapDispatchToProps)(PayBillContainer);
