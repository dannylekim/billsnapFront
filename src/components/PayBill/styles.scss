@import "../../variables";

.split-info {
  display: flex;
  justify-content: space-between;
}

.text__information {
  font-family: $primary-font;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  color: rgba(0, 0, 0, 0.71);
}

.small-input {
  //must override shards-react
  width: 15% !important;
}

.pay-button {
  margin-top: 1em;
}