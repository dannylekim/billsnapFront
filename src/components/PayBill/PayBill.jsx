import React from "react";
import {Button, FormInput} from "shards-react";
import "./styles.scss";

export default ({
  payBillHandler,
  onChangeHandler,
  billName,
  amountRemainingToPay,
}) => (
  <div>
    <div className="split-info text__information">
      <div>Bill Name</div>
      <div>{billName}</div>
    </div>
    <div className="split-info text__information">
      <div>Amount left to pay</div>
      <FormInput
          name={"amountRemaining"}
          value={amountRemainingToPay}
          onChange={onChangeHandler}
          className="small-input"
          step="0.01"
          type="number"
      />
    </div>
    <div className="split-info">
      <div />
      <Button
        outline
        pill
        theme="success"
        className="pay-button"
        onClick={payBillHandler}
      >
        Pay Bill
      </Button>
    </div>
  </div>
);
