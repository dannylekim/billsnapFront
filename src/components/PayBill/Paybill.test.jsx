import PayBill from "./PayBill";
import React from "react";

describe("PayBill", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("PayBill should match snap shot", () => {
        matches(<PayBill />);
      });
    });
  });
});
