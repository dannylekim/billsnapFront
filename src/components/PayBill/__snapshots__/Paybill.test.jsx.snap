// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayBill render snapshots ðŸ“¸ PayBill should match snap shot 1`] = `
<div>
  <div
    className="split-info text__information"
  >
    <div>
      Bill Name
    </div>
    <div />
  </div>
  <div
    className="split-info text__information"
  >
    <div>
      Amount left to pay
    </div>
    <input
      className="small-input form-control"
      step="0.01"
      type="number"
    />
  </div>
  <div
    className="split-info"
  >
    <div />
    <button
      className="pay-button btn btn-outline-success btn-pill"
      onClick={[Function]}
    >
      Pay Bill
    </button>
  </div>
</div>
`;
